### ISSUE TITLE
There ought to be a way to create covers for C structures/typedefs of the same name.

### ISSUE THREAD

**Issue by RomanHargrave** [fasterthanlime/rock#873]
I'm doing something with X11, the standard library for which uses naming conventions. This is a grave issue, as I need to create covers for the types I need to use (correct me if I am wrong). This presents an issue with some rather general names:

```ooc
Display: cover from Display
```

Which obviously will not work, which means I must decide whether to use some arbitrary internal name to refer to the `Display` structure (bad idea), or mangle the name of cover (i.e. `XDisplay`). I absolutely do *not* want to do either. Perhaps the following syntax could be used to disambiguate these sorts of things, as it is with functions:

```ooc
Display: cover from extern(Display)
```

**Comment by fasterthanlime** [comment]
That's not a *grave issue*, that's a minor annoyance :) Here's an example of how to do it: https://gist.github.com/fasterthanlime/ac6283a8ce34862a4ef4

tl;dr the syntax is:

```
Display: extern cover {
  // fields as usual
}
```
